### ISSUE TITLE
Wrong component listening on upload

### ISSUE THREAD

**Issue by craigward** [livewire/livewire#2103]
### Description
I'm implementing a file upload component that uses Filepond and Spatie media library. I'm still finishing up some functionality but have run into a weird bug.

Whats happening is I have two of the file upload components on the site, one for files and one for images. On the Livewire component I have 

```
@if($upload)
    <x-buttons.primary wire:click="save">Save</x-buttons.primary>
@endif
```

So once a file has been uploaded and a temp file created by Livewire the save button shows and the user can save.

However, with two components on the page, the save button shows up on the component that wasn't the one to initiate the upload. 

I've created a unique id for the ref on the Filepond component, and I've tried adding wire:key to things but the problem persists and I'm totally lost on this one.

Any help would be really appreciated

# Filepond component
```
@php
$pool = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
$length = 8;
$id = substr(str_shuffle(str_repeat($pool, 5)), 0, $length);
@endphp
<div wire:ignore x-data
     @refresh-filepond.window="FilePond.create( $refs.{{ $id }} );"
     x-init="
        FilePond.setOptions({
            server: {
                process:(fieldname, file, metadata, load, error, progress, abort, transfer, options) => {
                    @this.upload('{{ $attributes['wire:model'] }}', file, load, error, progress);
                },
                revert:(filename, load, error) => {
                    @this.removeUpload('{{ $attributes['wire:model'] }}', filename, load)
                }
            }
        });
        FilePond.create($refs.{{ $id }});
">
    <input type="file" x-ref="{{ $id }}">
</div>
```

# The Livewire component it sits inside
```
<x-cards.list title="{{ $title }}">
    <div>
        @foreach($items as $item)
            <x-cards.list-item>
                <button wire:click="viewImage()" class="truncate">{{ $item->file_name }}</button>

                <button wire:click="remove({{ $item->id }})">
                    <i class="fas fa-ban float-right text-danger"></i>
                </button>
            </x-cards.list-item>
        @endforeach
        <x-slot name="footerItems">
            <x-forms.filepond wire:model="upload" :key="$collectionName" />

            @if($upload)
                <x-buttons.primary wire:click="save">Save</x-buttons.primary>
            @endif
        </x-slot>
    </div>
</x-cards.list>
```

# How its being used on the site
```
<div>
    <livewire:files-relationship-manager :model="$course" collection-name="photos" title="Related images" fileType="images" />
</div>
<div>
    <livewire:files-relationship-manager :model="$course" collection-name="files" title="Related files" fileType="files" />
</div>
```

# The class

```
<?php

namespace App\Http\Livewire;

use Illuminate\View\View;
use Livewire\Component;
use Livewire\WithFileUploads;

class FilesRelationshipManager extends Component
{
    use withFileUploads;

    public $title = '';
    public $items = [];
    public $model = null;
    public $upload;
    public $collectionName = '';
    public $relationshipLookup = [
        'images' => ['jpg', 'jpeg', 'gif', 'png'],
        'files' => ['doc', 'docx', 'xls', 'xlsx', 'ppt', 'pptx', 'pdf'],
    ];

    protected $rules = [
        'upload' => 'nullable|max:1000',
    ];

    public function getItems()
    {
        $this->model->refresh();
        $this->items = $this->model->getMedia($this->collectionName);
    }

    public function save()
    {
        $this->model->addMediaFromDisk('/livewire-tmp/'.$this->upload->getFilename())->toMediaCollection($this->collectionName);
        $this->upload = null;
        $this->getItems();
        $this->dispatchBrowserEvent('refresh-filepond');
    }

    public function remove($filesId)
    {
        $this->model->deleteMedia($filesId);
        $this->getItems();
    }

    /**
     * Get the view / contents that represent the component.
     *
     * @return View|string
     */
    public function render()
    {
        $this->getItems();
        return view('livewire.files-relationship-manager', [
            'items' => $this->items
        ]);
    }
}
```

**Context**
- Livewire version: 2.3
- Laravel version: 8.10
- Alpine version: 2.7.1
- Browser: Firefox

**Comment by joshhanley** [comment]
@craigward can you label your blade code blocks with what their files names are? It's hard to follow which block of code is for what.

**Comment by craigward** [comment]
Added filenames which were added, but had been stripped

**Comment by rikschennink** [comment]
FilePond dev here. Not sure but `FilePond.setOptions` is global, and thus sets options for all instances. Maybe that's related to the issue. If you want to set options per instance you can pass them as a second parameter to `FilePond.create` 

```js
FilePond.create($refs.{{ $id }}, /* here */);
```

**Comment by craigward** [comment]
Thank you @rikschennink 

Moving the options into the create makes the component work exactly as it should

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)
